GVRf/Framework/framework/src/main/java/org/gearvrf/IAssetEvents.java

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2016 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gearvrf;
+
+/**
+ * Interface for processing asset load events.
+ * The asset loader raises an event whenever a texture
+ * or a model is loaded. It also raises an event when
+ * the entire asset (models and textures) is loaded.
+ * This interface lets you observe  these events.
+ */
+public interface IAssetEvents extends IEvents
+{
+    /**
+     * Called when a model is successfully loaded.
+     * This event will not be raised until the model and all of
+     * its textures are loaded.
+     * @param context   GVRContext used to load the model
+     * @param model     GVRSceneObject root of the model scene graph, null if model did not load.
+     * @param filePath  File path or URL of the model.
+     * @param errors    String with loading errors or null if successful
```

Comment:
I need to have these functions because the asset loader actually uses them internally. This is how it figures out whether all the textures loaded or not.

I don’t understand why having a parameter makes error handling harder. Can you explain that one.

My thinking behind the “errors” parameter for onAssetLoaded is for the end case. It is possible that the model or some of the textures do not load. Internally we might want to catch that case and do something about it. If you have no errors it implies your asset and everything it referenced loads correctly. If the model is null, it didn’t load. The errors string will have the accumulation of all the texture load errors.

From: rahul [mailto:notifications@github.com]
Sent: Thursday, June 2, 2016 6:00 PM
To: Samsung/GearVRf <GearVRf@noreply.github.com>
Cc: Nola Donato <nola.donato@samsung.com>; Author <author@noreply.github.com>
Subject: Re: [Samsung/GearVRf] Wait for textures to be loaded before adding model to the scene (#597)


In GVRf/Framework/framework/src/main/java/org/gearvrf/IAssetEvents.java<https://github.com/Samsung/GearVRf/pull/597#discussion_r65643796>:

> + * Interface for processing asset load events.

> + * The asset loader raises an event whenever a texture

> + * or a model is loaded. It also raises an event when

> + * the entire asset (models and textures) is loaded.

> + * This interface lets you observe  these events.

> + */

> +public interface IAssetEvents extends IEvents

> +{

> +    /**

> +     * Called when a model is successfully loaded.

> +     * This event will not be raised until the model and all of

> +     * its textures are loaded.

> +     * @param context   GVRContext used to load the model

> +     * @param model     GVRSceneObject root of the model scene graph, null if model did not load.

> +     * @param filePath  File path or URL of the model.

> +     * @param errors    String with loading errors or null if successful

Also having the errors param here makes it difficult to implement error handling. We would need to look at this value in addition onModelError and onTextureError which may not be ideal.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Samsung/GearVRf/pull/597/files/fb58b0721681e130ce48eec2b79e1fc44b7ecf78#r65643796>, or mute the thread<https://github.com/notifications/unsubscribe/AQw1SyCNbz1izN3X6hG3SHaHTImI5K_2ks5qH3yTgaJpZM4ItA3y>.